File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[201];


void dfs(int v, vector<vector<int>> computers){
    visited[v] = true;
    
    for(int i = 0; i<computers[v].size(); i++){
        if(!visited[i] && computers[v][i] == 1){
            dfs(i, computers);
            // visited[i] = false;
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int cnt = 0;
    for(int i = 0; i<n; i++){
        if(!visited[i]){
            dfs(i, computers);
            cnt++;
        }
    }
    
    return cnt;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    boolean [] visited = new boolean [201];
    
    public void bfs(int V, int [][] computers){
        Queue<Integer> q = new LinkedList<>();
        visited[V] = true;
        q.add(V);
        
        int next;
        
        while(!q.isEmpty()){
            int v = q.poll();
            
            for(int i = 0; i<computers.length; i++){
                if(i != v && computers[v][i] == 1 && !visited[i]){
                    visited[i] = true;
                    q.add(i);
                }
            }
        }
        
    }
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        
        for(int i = 0; i<n; i++){
            if(!visited[i]){
                answer++;
                bfs(i, computers);
            }
        }
        
        return answer;
    }
}